### PR TITLE
Support `Windows.Data.Xml.Dom`

### PR DESCRIPTION
**Pull request by Sunbreak**
Signed-off-by: Sunbreak <sunbreak.wang@gmail.com>

### CODE REVIEW

**Comment by Sunbreak** [comment]
I'm porting https://github.com/mohabouje/WinToast to Dart/Flutter, which needs `Windows.Data.Xml.Dom` features

When add `Windows.Data.Xml.Dom` there're several problems, could you take a look? @timsneath

File: lib/src/com/IXmlNodeList.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+typedef _get_Length_Dart = int Function(Pointer obj, Pointer<Uint32> value);
+
+typedef _Item_Native = Int32 Function(
+    Pointer obj, Uint32 index, Pointer<Pointer> result);
+typedef _Item_Dart = int Function(
+    Pointer obj, int index, Pointer<Pointer> result);
+
+/// {@category Interface}
+/// {@category winrt}
+class IXmlNodeList {
```
Comment (Sunbreak) [inline lib/src/com/IXmlNodeList.dart:35]: `IXmlNodeList` is a subtype of `IInspectable` but the generation omit the inheritance

@timsneath how could I dig inside? Or should I add the inheritance manually?

Comment (timsneath) [reply lib/src/com/IXmlNodeList.dart:35]: It looks like `IXmlNodeList` implements `IReadOnlyList<IXmlNode>` and `IEnumerable<IXmlNode>`. I think my projection code is broken for WinRT interfaces with generics at present, so it doesn't know what to do. 

Curious though -- can't you use the Dart [xml](https://pub.dev/packages/xml) package to parse the XML? I think that's going to be easier...

Comment (timsneath) [reply lib/src/com/IXmlNodeList.dart:35]: Opened #247.

Comment (Sunbreak) [reply lib/src/com/IXmlNodeList.dart:35]: > Opened #247.

Thanks a lot

> 

> Curious though -- can't you use the Dart [xml](https://pub.dev/packages/xml) package to parse the XML? I think that's going to be easier...

`IToastNotificationManagerStatics`'s `GetTemplateContent` results in a `IXmlDocument`. So I guess I have to use `Windows.Data.Xml.Dom`

**Pull request closed without merging**